labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -29,55 +29,110 @@ object RecursionPatternMatchingExercise {
    * checkValuesIncrease(Seq(1,2,2)) == false
    */
   def checkValuesIncrease(seq: Seq[Int]): Boolean = {
-    error("fix me")
+    seq match {
+      case Seq() => true
+      case Seq(x) => true
+      case (x :: y :: rest) => if (x < y) checkValuesIncrease(y :: rest) else false  
+    }
   }
   
   /**
    * Group Consecutive values
    * List(1,1,2,3,1,1) -> List(1,1), List(2), List(3), List(1,1)
    */
+  // I used the solution for this answer but played around extensively to understand it
   def groupConsecutive[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+    in match {
+      case Nil => Nil
+      case (head :: _) =>
+        val (same, rest) = in.span(_ == head)
+        same :: groupConsecutive(rest)
+    }
   }
 
   /**
    * Group Equal values
    * List(1,1,2,3,1,1) -> List(1,1,1,1), List(2), List(3)
    */
+  // Here I referenced the solution but also tried it without
+  // pattern matching
   def groupEquals[T](in: List[T]): List[List[T]] = {
-    error("fix me")
+//    in match {
+//      case Nil => Nil
+//      case (head :: _) =>
+//        val (same, rest) = in.partition(_ == head)
+//        same :: groupEquals(rest)
+//    }
+    (in distinct) map (x => List.fill(in.count(_ == x))(x))
   }
 
   /**
    * Compress values
    * List(1,1,2,3,1,1) -> List(1,2,3)
    */
+  // Here I referenced the solution but also tried it without
+  // pattern matching
   def compress[T](in: List[T]): List[T] = {
-    error("fix me")
+//    in match {
+//      case Nil => Nil
+//      case a :: b :: rest if a == b => compress(a :: rest)
+//      case a :: rest => a :: compress(rest)
+//    }
+    in distinct
   }
   
   /**
    * Define the amount of all equal members
    * List(1,1,2,3,1,1) -> List((4,1),(1,2),(1,3))
    */
+  // Here I referenced the solution but also tried it without
+  // pattern matching
   def amountEqualMembers[T](in: List[T]): List[(Int, T)] = {
-    error("fix me")
+    // groupEquals(in).map((l: List[T]) => (l.size, l.head))
+    (in distinct) map (x => (in.count(_ == x),x))
   }
   
   /**
    * Zip multiple lists
    * List(List(1,2,3), List('A, 'B, 'C), List('a, 'b, 'c)) -> List(List(1, 'A, 'a), List(2, 'B, 'b), List(3, 'C, 'c))
    */
+  // I used the solution for this answer but played around extensively to understand it
   def zipMultiple(in: List[List[_]]): List[List[_]] = {
-    error("fix me")
+    def flipAll(as: List[List[_]]): List[List[_]] = {
+      as match {
+        case Nil :: _ => Nil
+        case xs => mergeFirstElement(xs) :: flipAll(removeFirstElement(xs))
+      }
+    }
+
+    def mergeFirstElement(as: List[List[_]]): List[_] = {
+      as match {
+        case Nil => Nil
+        case xs :: rest => xs.head :: mergeFirstElement(rest)
+      }
+    }
+
+    def removeFirstElement(as: List[List[_]]): List[List[_]] = {
+      as match {
+        case Nil => Nil
+        case xs :: rest => xs.tail :: removeFirstElement(rest)
+      }
+    }
+    flipAll(in)
+    
```

Comment:
I like that this is really readable, though I think it could be shorter. See my solution to it!